libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/ui/RouteLineComponent.kt

```diff
@@ -117,19 +133,59 @@ class RouteLineComponent(
-                    navigationRoutes
+                    Pair(navigationRoutes, false)
                 } else if (!previewRoutes.isNullOrEmpty()) {
-                    previewRoutes
+                    Pair(previewRoutes, true)
                 } else {
-                    emptyList()
+                    Pair(emptyList(), false)
+                }
+            }.collect { pair ->
+                val routes = pair.first
+                val fromPreview = pair.second
+                if (fromPreview) {
+                    routeLineApi.setNavigationRoutes(routes) { value ->
+                        mapboxMap.getStyle()?.apply {
+                            routeLineView.renderRouteDrawData(this, value)
+                        }
+                    }
+                } else {
+                    routeLineApi.setNavigationRoutes(
+                        routes,
+                        mapboxNavigation.getAlternativeMetadataFor(routes)
+                    ) { value ->
+                        mapboxMap.getStyle()?.apply {
+                            routeLineView.renderRouteDrawData(this, value)
+                        }
+                    }
                 }
-            }.collect { routes ->
-                mapboxMap.getStyle()?.also { style ->
-                    val routeLines = routes.map { navigationRoute ->
-                        NavigationRouteLine(navigationRoute, null)
+            }
+        }
+
+        coroutineScope.launch {
+            mapboxNavigation.flowRouteAlternativeObserver().collect { alternatives ->
+                val currentRoutes = mapboxNavigation.getNavigationRoutes()
+                val primaryRoute = currentRoutes.firstOrNull()
+                ifNonNull(primaryRoute) { routePrimary ->
+                    val offBoardAlternatives = alternatives.filter {
+                        it.origin == RouterOrigin.Offboard
+                    }
+                    val updatedRoutes = arrayListOf(routePrimary)
+                    when {
+                        routePrimary.origin == RouterOrigin.Offboard -> {
+                            updatedRoutes.addAll(offBoardAlternatives)
+                        }
+                        offBoardAlternatives.isEmpty() -> {
+                            updatedRoutes.addAll(alternatives)
+                        }
+                        else -> {
+                            updatedRoutes.addAll(offBoardAlternatives)
+                        }
                     }
-                    val routeDrawData = routeLineApi.setNavigationRouteLines(routeLines)
-                    routeLineView.renderRouteDrawData(style, routeDrawData)
+                    /**
+                     * TODO: We should pass in currentLegIndex and invoke contractProvider.get().setRoutesWithIndex(mapboxNavigation, updatedRoutes, currentLegIndex)
+                     * val currentLegIndex = routeProgress.value?.currentLegProgress?.legIndex ?: 0
+                     */
```

Comment:
`RouteProgress` is passed to `NavigationRouteAlternativesObserver`, you can use it to retrieve the current leg index.

Reply:
@Zayankovsky This is fixed. Please take another look.